### PR TITLE
[3] Feat/result path

### PR DESCRIPTION
**Pull request by sam-n-johnston**
## Description

Fixes the ResultPath, issue #16

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Manual tests
- [x] Unit tests

### CODE REVIEW

File: src/StateProcessor.ts [hunk 1]
```diff
@@ -34,17 +35,47 @@ export class StateProcessor {
...
+      if (i === resultPathArray.length - 1) {
+        temp[key] = result;
+      } else {
+        if (!temp[key]) {
+          temp[key] = {};
+        }
+
+        temp = temp[key];
+      }
+    }
```
Comment (sam-n-johnston) [inline src/StateProcessor.ts:76]: Not great... but haven't thought of a better way for now.

**Comment by jefer590** [comment]
there's a conflict 😬

**Comment by sam-n-johnston** [comment]
I'm very confused by all these conflicts... there must be a difference in the branch on my local and this one here...

**Comment by sam-n-johnston** [comment]
@jefer590  resolved... but I guess there will be yet again conflicts on #24 ...

**Comment by jefer590** [comment]
yeah :/ 
probably the squash and merge? i'll try a pure merge 🤔

**Pull request merged**